test/command_test.exs

```diff
@@ -0,0 +1,14 @@
+defmodule Gnat.CommandTest do
+  use ExUnit.Case, async: true
+  alias Gnat.Command
+
+  test "formatting a simple unsub message" do
+    command = Command.build(:unsub, 12, []) |> IO.iodata_to_binary
+    assert command == "UNSUB 12\r\n"
+  end
+
+  test "formatting an unsub message with max messages" do
+    command = Command.build(:unsub, 12, [max_messages: 3]) |> IO.iodata_to_binary
```

Comment:
In erlang there is a generally accepted type called `iodata` (see [types and specs](http://erlang.org/doc/reference_manual/typespec.html)) which most network calls or other IO related functions accept. It can be a nested list of lists that contain binaries or character lists.

For simplicity in my test I am just converting back to binary so I can do a simple equality match, but we would actually pass this to the network as a list of small strings. It avoids allocating additional strings by the `UNSUB` and `\r\n` parts of the string will always be there so we only dynamically allocate the parts of the string that need to be dynamic.